### ISSUE TITLE
Can't install cdQA

### ISSUE THREAD

**Issue by JoaoPedroZucoloto** [cdqa-suite/cdQA#346]
ERROR: Could not find a version that satisfies the requirement torch==1.2.0 (from cdqa) (from versions: 0.1.2, 0.1.2.post1, 0.1.2.post2) ERROR: No matching distribution found for torch==1.2.0 (from cdqa)

**Comment by infact** [comment]
> ERROR: Could not find a version that satisfies the requirement torch==1.2.0 (from cdqa) (from versions: 0.1.2, 0.1.2.post1, 0.1.2.post2) ERROR: No matching distribution found for torch==1.2.0 (from cdqa)

Which OS are you using?

**Comment by infact** [comment]
Try this if on Windows

> pip install torch==1.2.0+cpu torchvision==0.4.0+cpu -f https://download.pytorch.org/whl/torch_stable.html

**Comment by infact** [comment]
Or if you are using anaconda, you can use the following command


`conda install pytorch==1.2.0 torchvision==0.4.0 -c pytorch`
--

**Comment by Kohimax** [comment]
I have face the below issue (windows environment)
"ERROR: Could not find a version that satisfies the requirement torch==1.2.0 (from cdqa) (from versions: 0.1.2, 0.1.2.post1, 0.1.2.post2)
ERROR: No matching distribution found for torch==1.2.0 (from cdqa)"

**Comment by Asma-droid** [comment]
ANy solution please ??

**Comment by circuitsacul** [comment]
Has anyone found a solution

**Comment by AlekseyKorshuk** [comment]
You just need to replace torch==1.2.0 to torch==1.5.0 in requirement.txt after cloning the directory. 
This works for me.

**Comment by sujsin77** [comment]
> Try this if on Windows
> 
> > pip install torch==1.2.0+cpu torchvision==0.4.0+cpu -f https://download.pytorch.org/whl/torch_stable.html

running above command gives following error
Looking in links: https://download.pytorch.org/whl/torch_stable.html
ERROR: Could not find a version that satisfies the requirement torch==1.2.0+cpu (from versions: 0.1.2, 0.1.2.post1, 0.1.2.post2, 1.4.0, 1.4.0+cpu, 1.4.0+cu92, 1.5.0, 1.5.0+cpu, 1.5.0+cu101, 1.5.0+cu92, 1.5.1, 1.5.1+cpu, 1.5.1+cu101, 1.5.1+cu92, 1.6.0, 1.6.0+cpu, 1.6.0+cu101, 1.7.0, 1.7.0+cpu, 1.7.0+cu101, 1.7.0+cu110, 1.7.1, 1.7.1+cpu, 1.7.1+cu101, 1.7.1+cu110, 1.8.0, 1.8.0+cpu, 1.8.0+cu101, 1.8.0+cu111)
ERROR: No matching distribution found for torch==1.2.0+cpu

**Comment by naveenvasou** [comment]
Anyone found a solution? I couldn't find one.

**Comment by Miru302** [comment]
> 
> 
> Anyone found a solution? I couldn't find one.

I encountered this problem as well, resolved by reinstalling 64bit version of python.

**Comment by angelopf** [comment]
Same issue

**Comment by NGeorgescu** [comment]
I know this is thread necromancy but [it looks like 1.2.0 is only compatible with python 3.7](https://pypi.org/project/torch/1.2.0/) according to the page, so you may need to either A) use a virtual environment to install 3.7 or B) download torch 1.2.0 and edit the setup file to allow you to install on 3.8.  The solution A is probably preferable considering that I believe 3.8 uses new keywords like "async" which break some packages, but YMMV.

**Comment by rockingkamal** [comment]
!pip install cdqa.  

ERROR: Could not find a version that satisfies the requirement cdqa (from versions: none)
ERROR: No matching distribution found for cdqa


any kind of information please share it ... Thanks in advance.

**Comment by NGeorgescu** [comment]
> !pip install cdqa.
> 
> ERROR: Could not find a version that satisfies the requirement cdqa (from versions: none) ERROR: No matching distribution found for cdqa
> 
> any kind of information please share it ... Thanks in advance.

Yeah it looks like it was pulled from pypi on a count of just being out of date.

You probably will get better mileage with LLMs. If you're dead-set on installing this package you should clone this repository and do `pip install .` (which pip installs the current directory)

**Comment by sujeets77** [comment]
Install cdqa manualy by following command
**git clone https://github.com/cdqa-suite/cdQA.git**
**cd cdQA**
Before installing change **torch version to 1.13.0 from 1.2.0** in requirements.txt file inside cdQA folder then run below command
**pip install -e .**

**Comment by MohdSiddiq12** [comment]
cloned the repo, cd cdqa and pip install -e . got this error then 
Obtaining file:///C:/Users/MohdS/OneDrive/Desktop/cdQA
  Preparing metadata (setup.py) ... done
Collecting Flask==1.1.1 (from cdqa==1.3.9)
  Using cached Flask-1.1.1-py2.py3-none-any.whl.metadata (4.4 kB)
Collecting flask_cors==3.0.8 (from cdqa==1.3.9)
  Using cached Flask_Cors-3.0.8-py2.py3-none-any.whl.metadata (5.4 kB)
Collecting joblib==0.13.2 (from cdqa==1.3.9)
  Using cached joblib-0.13.2-py2.py3-none-any.whl.metadata (4.5 kB)
Collecting pandas==0.25.0 (from cdqa==1.3.9)
  Using cached pandas-0.25.0.tar.gz (12.6 MB)
  Preparing metadata (setup.py) ... done
Collecting prettytable==0.7.2 (from cdqa==1.3.9)
  Using cached prettytable-0.7.2.zip (28 kB)
  Preparing metadata (setup.py) ... done
Collecting transformers==2.1.1 (from cdqa==1.3.9)
  Using cached transformers-2.1.1-py3-none-any.whl.metadata (30 kB)
Collecting scikit_learn==0.21.2 (from cdqa==1.3.9)
  Using cached scikit-learn-0.21.2.tar.gz (12.2 MB)
  Preparing metadata (setup.py) ... done
Collecting tika==1.19 (from cdqa==1.3.9)
  Using cached tika-1.19.tar.gz (24 kB)
  Preparing metadata (setup.py) ... done
INFO: pip is looking at multiple versions of cdqa to determine which version is compatible with other requirements. This could take a while.
ERROR: Could not find a version that satisfies the requirement torch==1.5.0 (from cdqa) (from versions: 1.11.0, 1.12.0, 1.12.1, 1.13.0, 1.13.1, 2.0.0, 2.0.1, 2.1.0, 2.1.1, 2.1.2, 2.2.0, 2.2.1)
ERROR: No matching distribution found for torch==1.5.0

**Comment by NGeorgescu** [comment]
@MohdSiddiq12 : so, looking at the error, it means that your version of python can't find a version of torch that matches.  If you look at the page for torch [here](https://pypi.org/project/torch/), it shows in the sidebar that you need python >= 3.8.0 

I imagine that your python version is probably 3.7 or earlier... OR you're accidentally running a python2 pip install or something.

**Comment by MohdSiddiq12** [comment]
@NGeorgescu  i downgraded my python version using virtual environment as it was suggested that it might work in python previous version, my current version of python is 3.10 
and after spending 2+ hours brainstorming to understand why it is not being installed and after spending time with various ai tools got to know that cdqa lib doesnt work in windows, it only works in ubuntu and linux

**Comment by Dinesh4242** [comment]
ERROR: Could not find a version that satisfies the requirement cdQA (from versions: none)
ERROR: No matching distribution found for cdQA
